spec/support/features/course_details_steps.rb

```diff
@@ -49,6 +54,11 @@ def and_the_course_details_is_marked_completed
       expect(review_draft_page).to have_course_details_completed
     end
 
+    def and_the_course_study_mode_field_is_completed
+      course_details_page.study_mode_full_time&.choose
+    rescue Capybara::ElementNotFound # rubocop:disable Lint/SuppressedException
```

Comment:
If the element can't be found, could there be something wrong with the test setup itself? It feels weird to suppress this error when it could be trying to point out something meaningful

Reply:
I would normally check if the trainee is the type that requires study mode, but in this case, we don't actually have a proper `trainee` because of how the tests have been refactored.
Visually we only show the element if it's a trainee that requires it.

Reply:
So i think this step is used as part of the end to end journeys. I don't think we want to do any suppressions for these specs as they're designed to catch out any weird data issues for that specific route. Could we not test the the study mode behaviour in a different step and then use that for the end to end journeys which require it? (which i think at the moment are just the `assessment_only` `pg_teaching_apprenticeship` end to end specs).

Reply:
Change this to let the tests pass in whether they expect `study_mode` as a param:
`and_the_course_details_is_complete(requires_study_mode: true)`

Reply:
Nice. Do we need to use it in the following specs? eg add `and_the_course_details_is_complete(requires_study_mode: true)` in the following specs:

- spec/features/end_to_end/provider_led_journey_spec.rb
- spec/features/end_to_end/school_direct_salaried_journey_spec.rb
- spec/features/end_to_end/school_direct_tuition_fee_journey_spec.rb

If i understood correctly, they're some of the routes which require study mode?

Reply:
Nice. Do we need to also add it to `spec/features/end_to_end/provider_led_journey_spec.rb`?

Reply:
Yes, I've added the change to those tests.
Provider led journey uses publish course, that's a different card:
https://trello.com/c/XBDN1FnL/2328-l-course-study-mode-question-publish-trainees